hex.go

```diff
@@ -1,33 +1,154 @@
 package hex
 
 import (
+	"github.com/oleiade/reflections"
+	"gopkg.in/yaml.v3"
+	"io/ioutil"
+	"log"
 	"math/rand"
+	"strconv"
+	"time"
 )
 
+type HexFlowerContent struct {
+	Start     int    `yaml:"start"`
+	Hexes     struct {
+		Num1  string `yaml:"1"`
+		Num2  string `yaml:"2"`
+		Num3  string `yaml:"3"`
+		Num4  string `yaml:"4"`
+		Num5  string `yaml:"5"`
+		Num6  string `yaml:"6"`
+		Num7  string `yaml:"7"`
+		Num8  string `yaml:"8"`
+		Num9  string `yaml:"9"`
+		Num10 string `yaml:"10"`
+		Num11 string `yaml:"11"`
+		Num12 string `yaml:"12"`
+		Num13 string `yaml:"13"`
+		Num14 string `yaml:"14"`
+		Num15 string `yaml:"15"`
+		Num16 string `yaml:"16"`
+		Num17 string `yaml:"17"`
+		Num18 string `yaml:"18"`
+		Num19 string `yaml:"19"`
+	} `yaml:"hexes"`
+}
 var neighbors = map[int][]int{
-	9: []int{10, 16, 13, 8, 4, 6},
+	1: {2,5,3,0,0,0},
+	2: {4,7,5,1,0,0},
+	3: {5,8,6,0,0,1},
+	4: {0,9,7,2,0,0},
+	5: {7,10,8,3,1,2},
+	6: {8,11,0,0,0,3},
+	7: {9,12,10,5,2,4},
+	8: {10,13,11,6,3,5},
+	9: {0,14,12,7,4,0},
+	10: {12,15,13,8,5,7},
+	11: {13,16,0,0,6,8},
+	12: {14,17,15,10,7,9},
+	13: {15,18,16,11,8,10},
+	14: {0,0,17,12,9,0},
+	15: {17,19,18,13,10,12},
+	16: {18,0,0,0,11,13},
+	17: {0,0,19,15,12,14},
+	18: {19,0,0,16,13,15},
+	19: {0,0,0,18,15,17},
 }
 
+var NavHex = map[int]int{
+	2: 2,
+	3: 2,
+	4: 3,
+	5: 3,
+	6: 4,
+	7: 4,
+	8: 5,
+	9: 5,
+	10: 0,
+	11: 0,
+	12: 1,
+}
 
 type Flower struct {
 	currentNode int
 	content map[int]string
+	navHex map[int]int
+	Random *rand.Rand
 }
 
-func NewFlower(content map[int]string) *Flower{
+func NewFlower(content map[int]string, nh map[int]int, start int) *Flower{
 	return &Flower{
-		currentNode: 9,
+		currentNode: start,
 		content: content,
+		navHex: nh,
+		Random: rand.New(rand.NewSource(time.Now().Unix())),
 	}
 }
 
-func (f *Flower) GoToNext() {
+func LoadContent(filename string) (map[int]string, int){
+	var hfc HexFlowerContent
+	yamlFile, err := ioutil.ReadFile(filename)
+	if err!=nil {
+		log.Fatalf("can't open file: %v", err)
```

Comment:
It probably makes sense for this function to return error, doesn't it? In general, any function that _can_ error should return an error value rather than, for example, panicking, calling `log.Fatal`, or otherwise terminating the program. My rule of thumb is that only `func main` has the right to terminate the program! All other functions should refer that decision back to `main`.